File: backend/src/main/java/de/neuefische/capstone/backend/model/MonthlyBalance.java
package de.neuefische.capstone.backend.model;

import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class MonthlyBalance {

    private final String month;
    private final BigDecimal totalIncome;
    private final BigDecimal totalExpenses;
    private final BigDecimal fixedCosts;
    private final BigDecimal variableCosts;
    private final BigDecimal oneTimeCosts;
    private final BigDecimal balance;

    private final List<Entry> monthlyEntries;
}

File: frontend/src/model/MonthlyBalance.ts
import {Entry} from "./Entry.ts";

export type MonthlyBalance = {
    yearMonth: string;
    totalIncome: string;
    totalExpenses: string;
    fixedCosts: string;
    variableCosts: string;
    oneTimeCosts: string;
    balance: string;
    monthlyEntries: Entry [];
}

export type MonthlyBalanceAmounts = {
    totalIncome: string;
    totalExpense: string;
    fixedCosts: string;
    variableCosts: string;
    oneTimeCosts: string;
    balance: string;
}